### PR TITLE
reference/sql: add document for the option 'allow-expression-index'  which enable creating expression index

### PR DESCRIPTION
**Pull request by wjhuang2016**
### What is changed, added or deleted? (Required)
Add document for the option 'allow-expression-index'  which enables creating expression index.

### Which TiDB version(s) do your changes apply to? (Required)

- [x] master (the latest development version, including v4.0 changes for now)
- [ ] v3.1 (TiDB 3.1 versions)
- [ ] v3.0 (TiDB 3.0 versions)
- [ ] v2.1 (TiDB 2.1 versions)

**If you select two or more versions from above**, to trigger the bot to cherry-pick this PR to your desired release version branch(es), you **must** add corresponding labels such as **needs-cherry-pick-3.1**, **needs-cherry-pick-3.0**, and **needs-cherry-pick-2.1**.

### What is the related PR or file link(s)?
https://github.com/pingcap/tidb/pull/15259

- This PR is translated from:
- Other reference link(s):

### CODE REVIEW

File: reference/sql/statements/create-index.md [hunk 1]
```diff
@@ -130,6 +130,14 @@ Query OK, 0 rows affected (0.31 sec)
 
 ## 表达式索引
 
+表达式索引目前是一个实验特性，如果需要使用，需要在 config 文件中设置
```
[the remark this answers is not in the document]
Comment (wjhuang2016) [reply reference/sql/statements/create-index.md]: Yeah，I will add it soon.

**Comment by Joyinqin** [comment]
outdated changes

**Pull request merged**